File: program/cpp/Vehicle.cpp
/*Saya Muhammad Muhammad Fadlul Hafiizh [2209889] mengerjakan soal latprak_4 dalam mata kuliah DPBO.
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan, Aamiin */
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>

// Deklarasi namespace
using namespace std;

//class vehicle ini menjadi kelas induk dari class motorcycle dan car dimana seperti pada dunia nyata motor dan mobil adalah bagian keluarga dari kendaraan
//dan semua atribut kelas ini general seperti plat no, merk yang intinya pasti akan dimiliki oleh motor ataupun mobil (sebuah kendaraan)
class Vehicle
{
private:
    string plat_no;
    string merk;
    string tahun_produksi;
    string warna;

public:
    Vehicle(/* args */){}
    //Overloading constructor
    Vehicle(string plat_no, string merk, string tahun_produksi, string warna){
        this->plat_no = plat_no;
        this->merk = merk;
        this->tahun_produksi = tahun_produksi;
        this->warna = warna;
    }

    //enakpsulasi semua atribut
    string getPlat_no() {
        return this->plat_no;
    }
    void setPlat_no(string plat_no) {
        this->plat_no = plat_no;
    }


    string getMerk() {
    	return this->merk;
    }
    void setMerk(string merk) {
    	this->merk = merk;
    }


    string getTahun_produksi() {
    	return this->tahun_produksi;
    }
    void setTahun_produksi(string tahun_produksi) {
    	this->tahun_produksi = tahun_produksi;
    }


    string getWarna() {
    	return this->warna;
    }
    void setWarna(string warna) {
    	this->warna = warna;
    }

    //desctructor
    ~Vehicle(){}
};

File: python/program/Vehicle.py
#Saya Muhammad Muhammad Fadlul Hafiizh [2209889] mengerjakan soal latprak_4 dalam mata kuliah DPBO.
#untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan, Aamiin */

#class vehicle ini menjadi kelas induk dari class motorcycle dan car dimana seperti pada dunia nyata motor dan mobil adalah bagian keluarga dari kendaraan
#dan semua atribut kelas ini general seperti plat no, merk yang intinya pasti akan dimiliki oleh motor ataupun mobil (sebuah kendaraan)
class Vehicle:
    def __init__(self, plat_no="", merk="", tahun_produksi="", warna=""):
        self.__plat_no = plat_no
        self.__merk = merk
        self.__tahun_produksi = tahun_produksi
        self.__warna = warna

    #enakpsulasi semua atribut
    def get_plat_no(self):
        return self.__plat_no

    def set_plat_no(self, plat_no):
        self.__plat_no = plat_no

    def get_merk(self):
        return self.__merk

    def set_merk(self, merk):
        self.__merk = merk

    def get_tahun_produksi(self):
        return self.__tahun_produksi

    def set_tahun_produksi(self, tahun_produksi):
        self.__tahun_produksi = tahun_produksi

    def get_warna(self):
        return self.__warna

    def set_warna(self, warna):
        self.__warna = warna
